moment-element.html

```diff
@@ -33,80 +33,77 @@
          */
-        inputFormat: String,
+        inputFormat: {
+          type: String,
+          value: null
+        },
 
         /**
          * The datetime output format. Options are 'now' or datetime using the
          * [moment Format](http://momentjs.com/docs/#/displaying/format/).
          */
-        outputFormat: String,
+        outputFormat: {
+          type: String,
+          value: null
+        },
 
         /**
          * Relative time using [momen time from now](http://momentjs.com/docs/#/displaying/fromnow/)
          * or [momen Time from datetime](http://momentjs.com/docs/#/displaying/from/).
          */
-        from: String,
+        from: {
+          type: String,
+          value: null
+        },
 
         /**
          * Relative time using [momen Time to now](http://momentjs.com/docs/#/displaying/tonow/)
          * or [momen Time to datetime](http://momentjs.com/docs/#/displaying/to/).
          */
-        to: String,
+        to: {
+          type: String,
+          value: null
+        },
 
         /**
          * The output datetime.
          */
-        output: String
-      },
-
-      ready: function() {
-        this.output = this._initializeMoment();
-
-        if (this.from) {
-          this.output = this._getFrom();
-        } else if (this.to) {
-          this.output = this._getTo();
+        output: {
+          type: String,
+          computed: 'computeOutput(datetime, inputFormat, outputFormat, from, to)'
         }
-
-        Polymer.dom(this.root).innerHTML = this.output;
       },
 
       _initializeMoment: function() {
-        if (this.datetime && this.inputFormat) {
-          var output = moment(this.datetime, this.inputFormat);
-        } else if (this.datetime) {
-          var output = moment(this.datetime);
-        } else {
-          var output = moment();
-        }
-
-        if (this.outputFormat) {
-          return output.format(this.outputFormat);
-        } else {
-          return output;
-        }
+        var output = moment(this.datetime, this.inputFormat);
+        return this.outputFormat ? output.format(this.outputFormat) : output;
       },
 
-      _getFrom: function() {
-        if (this.from === 'now') {
-          return this.output.fromNow();
-        } else {
-          return this.output.from(moment(this.from));
-        }
+      _getFrom: function(from, output) {
+        return from === 'now' ? output.fromNow() : output.from(moment(from));
       },
 
-      _getTo: function() {
-        if (this.to === 'now') {
-          return this.output.toNow();
-        } else {
-          return this.output.to(moment(this.to));
+      _getTo: function(to, output) {
+        return to === 'now' ? output.toNow() : output.to(moment(to));
+      },
+      
+      computeOutput: function(datetime, inputFormat, outputFormat, from, to) {
+        var output = this._initializeMoment();
+        if (from) {
```

Comment:
I'm not sure... I have re-read it and I think we could have just:

```javascript
_computeOutput: function(datetime, inputFormat, outputFormat, from, to) {
  var output = inputFormat ? moment(datetime, inputFormat) : moment(datetime);

  output = outputFormat ? output.format(outputFormat) : output;

  if (from) {
    output = from === 'now' ? output.fromNow() : output.from(moment(from));
  } else if (to) {
    output = to === 'now' ? output.toNow() : output.to(moment(to));
  }

  return output;
}
```

Only 11 lines. :)